### PR TITLE
fix(search): fix outline appearing onClick around the X icon - FE-3999

### LINKED ISSUE #3857: Search "X" icon outline appears onClick
### Current behaviour
When clicking the `X` icon to clear the input in the Search component, a golden outline quickly appears.

https://user-images.githubusercontent.com/56251247/112610166-4c911a00-8e14-11eb-9e76-0522b7ef5785.mov

### Expected behaviour
When clicking the "X" icon to clear the input, no outline should appear around the "X" icon.

### Suggested solution(s)
Will probably need to use a flag to control when the icon can have a focus outline.
Solution will also need to make sure that when using keyboard navigation that the "X" can still appear focused.

### Your environment

| Software         | Version(s) |
| ---------------- | ---------- |
| carbon-react     | 68.5.1
| carbon-factory   |
| react-scripts    | 
| React            |
| Browser          |
| npm              |
| Operating System |

### PR DESCRIPTION
**Pull request by DipperTheDan**
Fixes a styling bug where the X icon appears a golden outline onClick

fixes #3857

### Proposed behaviour
Focus outline should only appear around 'X' icon when icon is navigated to via the keyboard.

### Current behaviour
Focus outline appears onClick.

### Checklist

- [x] Commits follow our style guide
<del> - [ ] Screenshots are included in the PR if useful </del>
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
<del> - [ ] Storybook added or updated if required </del>
<del> - [ ] Typescript `d.ts` file added or updated if required </del>
- [x] Carbon implementation and Design System documentation are congruent

### Additional context
N/A

### Testing instructions
https://codesandbox.io/s/carbon-quickstart-forked-dojtr?file=/src/index.js

### CODE REVIEW

File: src/__experimental__/components/search/search.component.js [hunk 1]
```diff
@@ -14,6 +14,7 @@ const Search = ({
   defaultValue,
   onChange,
   onClick,
+  onMouseDown,
```
Comment (edleeks87) [inline src/__experimental__/components/search/search.component.js:17]: this prop is never used, I'm not sure we need it tbh as we only want the mouseDown event handler to stop the focus behaviour

File: src/__experimental__/components/search/search.component.js [hunk 2]
```diff
@@ -178,13 +184,16 @@ Search.propTypes = {
   /** Prop for `onClick` events.
    * `onClick` events are triggered when the `searchButton` is clicked */
   onClick: PropTypes.func,
+  /** Prop for `onMouseDown` events.
```
Comment (edleeks87) [inline src/__experimental__/components/search/search.component.js:187]: see comment above

File: src/__experimental__/components/search/search.d.ts [hunk 3]
```diff
@@ -9,13 +9,17 @@ export interface SearchProps {
    * `onClick` events are triggered when the `searchButton` is clicked
    */
   onClick?: (ev: React.SyntheticEvent) => void;
+    /** Prop for `onMouseDown` events.
```
Comment (edleeks87) [inline src/__experimental__/components/search/search.d.ts:12]: This is why your linter has failed fyi, I also think we don't need to surface this prop

File: src/__experimental__/components/search/search.component.js (hunk 1 above)
Comment (DipperTheDan) [reply src/__experimental__/components/search/search.component.js:17]: It's gone on the next commit 👍🏻

File: src/__experimental__/components/search/search.component.js (hunk 2 above)
Comment (DipperTheDan) [reply src/__experimental__/components/search/search.component.js:187]: Consider it gone.

File: src/__experimental__/components/search/search.d.ts (hunk 3 above)
Comment (DipperTheDan) [reply src/__experimental__/components/search/search.d.ts:12]: It's gone on the next commit.

**Review by edleeks87 — APPROVED** [review]

**Review by sjohnson-sage — APPROVED** [review]

**Review by sjohnson-sage — APPROVED** [review]

**Comment by tempertemper** [comment]
I've had a quick look at this with @harpalsingh (who'll get back to you separately with approval/etc.). One thing I noticed as I poked around the generated code was that we're using a `<span>` rather than the `<button>` element. I'll raise something separately about getting this switched as we've gone to quite a lot of effort to replicate the functionality of a `<button>` (keyboard behaviour, tab indexed, etc.) but a button would give us all of that (and more) for free (we're missing a `role="button"` that would be implicit with a `<button>` element, so that screen reader users would know what the element is and speech recognition users could interact with it as a button).

**Comment by harpalsingh** [comment]
We are happy on the UX and A11y, but just a note that @tempertemper will be looking into possible updates to the element tag, whether it might need to be <button> but that will be raised separately.

**Review by ZhuoyuJin — DISMISSED** [review]
Tested in Chrome, Edge, Firefox, and Safari. No issues found, testing passed

**Review by ZhuoyuJin — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Pull request merged**